assets/js/util/insufficient-permissions-error-description.js

```diff
@@ -26,73 +26,164 @@ import { __, sprintf } from '@wordpress/i18n';
  *
  * @since 1.16.0
  *
- * @param {string} error  Original error message.
- * @param {Object} module Module data.
- * @return {string} Error description.
+ * @param {string} error              Original error message.
+ * @param {Object} module             Module data.
+ * @param {string} module.name        The name of the module.
+ * @param {string} module.slug        The slug of the module.
+ * @param {Object} module.owner       The owner of the module.
+ * @param {string} module.owner.login The The login of the current owner.
+ * @return {string}                   Error description.
  */
-export function getInsufficientPermissionsErrorDescription( error, module ) {
-	const { slug = '', name = '', owner = null } = module || {};
+export function getInsufficientPermissionsErrorDescription(
+	error = '',
+	module = {}
+) {
+	const { slug = '', name = '', owner = {} } = module || {};
 
 	// If no module data provided, it is impossible to provide a more clear message.
 	if ( ! slug || ! name ) {
 		return error;
 	}
 
-	let message = '';
-	let userInfo = '';
+	let messages;
+	switch ( slug ) {
+		case 'analytics':
+			messages = analyticsError( error );
+			break;
+		case 'search-console':
+			messages = searchConsoleErrors();
+			break;
+		default:
+			messages = [];
+			break;
+	}
+
+	if ( messages.length === 0 ) {
+		messages = emptyError( name );
+	}
+
+	messages = [ ...messages, ...userLoginErrors( owner ) ];
+
+	//  Create a full sentence separated by full stops "." and a space to separate 2 messages, if more than one message
+	const sentence = messages.join( '. ' );
+
+	/**
+	 * If the message contains less than 2 sentences return the sentence as it is, otherwise add a full stop at the
+	 * end of the sentence, in order to follow the UX guidelines.
+	 *
+	 * "text with 1 or 2 sentences should not end in a full stop".
+	 *
+	 * @since n.e.x.t
```

Comment:
Looks like this one fails an `eslint` rule if we remove it, are we okay with that being the case ?

Reply:
@mitogh just use `//` comments and `eslint` won't require `@since` tag:

```js
// If the message contains less than 2 sentences return the sentence as it is, otherwise add a full stop at the
// end of the sentence, in order to follow the UX guidelines.
//
// "text with 1 or 2 sentences should not end in a full stop".
//
// @see https://github.com/google/site-kit-wp/issues/4160
return messages.length <= 2 ? sentence : `${ sentence }.`;
```

Reply:
> @mitogh just use `//` comments and `eslint` won't require `@since` tag:

I think we could also use a block comment using a single `*` at the start rather than two, although we have multiple lines with `//` as well 😄 

```js
/*
 * something without needing @since should work too
 */
```